playground/backend/internal/executors/executor_builder.go

```diff
@@ -82,87 +82,97 @@ func (b *ExecutorBuilder) WithTestRunner() *UnitTestExecutorBuilder {
 	return &UnitTestExecutorBuilder{*b}
 }
 
-//WithCommand adds compile command to executor
+// WithCommand adds compile command to executor
 func (b *CompileBuilder) WithCommand(compileCmd string) *CompileBuilder {
 	b.actions = append(b.actions, func(e *Executor) {
 		e.compileArgs.commandName = compileCmd
 	})
 	return b
 }
 
-//WithWorkingDir adds dir path to executor
+// WithWorkingDir adds dir path to executor
 func (b *CompileBuilder) WithWorkingDir(dir string) *CompileBuilder {
 	b.actions = append(b.actions, func(e *Executor) {
 		e.compileArgs.workingDir = dir
 	})
 	return b
 }
 
-//WithArgs adds compile args to executor
+// WithArgs adds compile args to executor
 func (b *CompileBuilder) WithArgs(compileArgs []string) *CompileBuilder {
 	b.actions = append(b.actions, func(e *Executor) {
 		e.compileArgs.commandArgs = compileArgs
 	})
 	return b
 }
 
-//WithFileName adds file name to executor
+// WithFileName adds file name to executor
 func (b *CompileBuilder) WithFileName(fileName string) *CompileBuilder {
+	return b.WithFileNames([]string{fileName})
+}
+
+// WithFileNames adds file names to executor
+func (b *CompileBuilder) WithFileNames(fileNames []string) *CompileBuilder {
 	b.actions = append(b.actions, func(e *Executor) {
-		e.compileArgs.fileName = fileName
+		e.compileArgs.fileNames = fileNames
 	})
 	return b
 }
```

Comment:
What about just one func that handles both cases?:
```
func (e *CompileBuilder) WithFileNames(fileNames ...string) *CompileBuilder {
	e.compileArgs.fileNames = fileNames
        return e
}
```

https://go.dev/play/p/IQ59C-t8wNA